### PR TITLE
User Registration 

### LINKED ISSUE #22: User Registration
1. Login:
   - The application must provide users with the ability to log in
   - Users must be able to log in using their email and password
2. User registration:
   - The application must include an option in the menu for users to register
   - During registration, users must be asked for their name, email and password

### PR DESCRIPTION
**Pull request by angel23v**
Use devise gem to authenticate users, custom views generated & modify some behaviors on navbar

For more info see: https://github.com/BrightCoders-Institute/s5a1-administrador-de-eventos-team4/issues/22

### CODE REVIEW

File: app/views/devise/confirmations/_navbar.hml.erb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    <div class="collapse navbar-collapse" id="navbarSupportedContent">
+      <ul class="navbar-nav me-auto mb-2 mb-lg-0">
+        <li class="nav-item">
+          <%= link_to "Iniciar", events_index_path , class: "btn btn-dark text-light"%>
+        </li>
+      </ul>
+      <form class="d-flex" role="search">
+        <input class="form-control me-2" type="search" placeholder="Buscar Evento" aria-label="Search">
+        <button class="btn btn-outline-success" type="submit">Buscar</button>
+      </form>
```
Comment (MauricioNC) [inline app/views/devise/confirmations/_navbar.hml.erb]: Just as suggestion, you can try in another PR create the search functionality with Rails

Comment (angel23v) [reply app/views/devise/confirmations/_navbar.hml.erb]: In fact, we chose to eliminate that option 'cause we saw that we will later use a filter. 
We forgot to delete it in some files but surely we will create another PR.
Thank you.

File: app/views/devise/sessions/_navbar.html.erb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
+      <span class="navbar-toggler-icon"></span>
+    </button>
+    <div class="collapse navbar-collapse" id="navbarSupportedContent">
+      <ul class="navbar-nav me-auto mb-2 mb-lg-0">
+        <li class="nav-item">
+          <% if signed_in? && current_page?(events_main_path)%>
+            <%= link_to "Iniciar", events_index_path , class: "btn btn-dark text-light"%>
+          <% end %>
+        </li>
```
Comment (MauricioNC) [inline app/views/devise/sessions/_navbar.html.erb]: What happens if the user is not signed in or the current page is not equals to events_main_path?.
If the condition is not met do you'll leave an empty li tag?

Same for the other li tag and for the app/views/events/_navbar.html.erb file

**Review by MauricioNC — CHANGES_REQUESTED** [review]

**Review by juperameza — CHANGES_REQUESTED** [review]
Remove the navbar file and just add it in the layout directory

**Comment by juperameza** [comment]
> Remove the navbar file and just add it in the layout directory

And do everything in a single language

File: config/routes.rb [hunk 3]
```diff
@@ -17,6 +18,70 @@
...
+    get '/users/sig_in' => 'devise/sessions#new'
+  end
+
+  devise_scope :user do
+    post '/users/sign_in' => 'devise/sessions#create'
+  end
+
+  devise_scope :user do
+    get '/users/sign_out' => 'devise/sessions#destroy'
+  end
```
Comment (juperameza) [inline config/routes.rb]: Why are you doing this

**Review by juperameza — CHANGES_REQUESTED** [review]

File: config/routes.rb (hunk 3 above)
Comment (juperameza) [reply config/routes.rb]: and in multiple lines too

**Review by juperameza — APPROVED** [review]

**Review by MauricioNC — APPROVED** [review]

**Pull request merged**